### PR TITLE
split temperature protection into heating and cooling

### PR DESCRIPTION
**Pull request by EmoGarbage404**
## About the PR

Allows temperature protection component to specify separate coefficients based on whether or not the player is warming up or cooling down.

adjusts some items (firesuits and winter clothing) to have more reasonable temperature protections for the different directions.

## Why / Balance

There's a lot of times where balancing insulation equally between heating and cooling has adverse effects. Especially for specialty temperature  protection (cold prot, heat prot) having the coefficient applied equally makes them incredibly strong and generally useful.

splitting it up like this lets us have more control over how we apply the modifiers and thus better balance firesuits and other items.

## Requirements

- [x] I have read and I am following the [Pull Request Guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html). I understand that not doing so may get my pr closed at maintainer’s discretion
- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

**Changelog**

:cl:
- tweak: Firesuits are now worse at keeping in heat and winter clothes make you get warmer quicker.

### CODE REVIEW

**Review by slarticodefast — CHANGES_REQUESTED** [review]
Ok, from a physics perspective the conductivity *should* always be the same in both directions. I tried thinking of ways this can be abused, but could not come up with anything. From a gameplay perspective this change is great.

**Comment by EmoGarbage404** [comment]
> Ok, from a physics perspective the conductivity should always be the same in both directions

i be hating on the laws of physics. they cramp my ephemeral style.

File: Content.Server/Temperature/Components/TemperatureProtectionComponent.cs [hunk 1]
```diff
@@ -7,10 +7,16 @@ namespace Content.Server.Temperature.Components;
...
     /// <summary>
-    ///     How much to multiply temperature deltas by.
+    ///     How much to multiply temperature deltas by heating up
     /// </summary>
     [DataField]
-    public float Coefficient = 1.0f;
+    public float HeatingCoefficient = 1.0f;
+
+    /// <summary>
+    ///     How much to multiply temperature deltas by when cooling down
```
[the remark this answers is not in the document]
Comment (EmoGarbage404) [reply Content.Server/Temperature/Components/TemperatureProtectionComponent.cs:16]: i dont really think it's worth having breaking changes there for something that really doesn't matter in the grand scheme of things. generally it's better not to unnecessarily rename classes and variables.

[the remark this answers is not in the document]
Comment (slarticodefast) [reply Content.Server/Temperature/Components/TemperatureProtectionComponent.cs]: They are both only used for this single thing, so it should be fine to rename them. Not really necessary, but makes it much more unsterstandable in my opinion

**Review by slarticodefast — APPROVED** [review]
The test fail is a heisentest, I had that one fail randomly a few times before.

**Pull request merged**